File: Python/formulas.py
import math

def convertToCelsius(temp):
	return (temp - 32) * 0.5555555 # rough estimate

def convertToFahrenheit(temp):
	return (temp * 1.8) + 32

def quadraticPlus(a, b, c):
	return (-b + math.sqrt(math.pow(b,2) - (4 * a * c))) / (2 * a)

def quadraticMinus(a, b, c):
	return (-b - math.sqrt(math.pow(b,2) - (4 * a * c))) / (2 * a)

class Point(object):

	def __init__(self, x, y):
		self.x = x
		self.y = y

	def __str__(self):
		return "(" + str(self.x) + ", " + str(self.y) + ")"

def distance(p1, p2):
	return math.sqrt(math.pow((p2.x - p1.x),2) + math.pow((p2.y - p1.y),2))

def midpoint(p1, p2):
	return Point((p1.x + p2.x) / 2, (p1.y + p2.y) / 2)

def slopeIntercept(p, yIntercept):
	return (p.y - yIntercept) / p.x

def slope(p1, p2):
	return (p2.y - p1.y) / (p2.x - p1.x)

def pythagoreanTheorem(a, b):
	return math.sqrt(math.pow(a,2) + math.pow(b,2))

File: Kotlin/Formulas.kt
import kotlin.math.pow
import kotlin.math.sqrt

fun convertToCelsius(temp: Double): Double {
    return (temp - 32) * 0.5555555 // rough estimate
}

fun convertToFahrenheit(temp: Double): Double {
    return (temp * 1.8) + 32
}

fun quadraticPlus(a: Double, b: Double, c: Double): Double {
    return (-b + sqrt(b.pow(2) - (4 * a * c))) / (2 * a)
}

fun quadraticMinus(a: Double, b: Double, c: Double): Double {
    return (-b - sqrt(b.pow(2) - (4 * a * c))) / (2 * a)
}


class Point (var x: Double, var y: Double) {
    override fun toString(): String {
        return "($x, $y)"
    }
}

fun distance(p1: Point, p2: Point): Double {
    return sqrt((p2.x - p1.x).pow(2) + (p2.y - p1.y).pow(2))
}

fun midpoint(p1: Point, p2: Point): Point {
    return Point((p1.x + p2.x) / 2, (p1.y + p2.y) / 2)
}

fun slopeIntercept(p: Point, yIntercept: Double): Double {
    return (p.y - yIntercept) / p.x
}

fun slope(p1: Point, p2: Point): Double {
    return (p2.y - p1.y) / (p2.x - p1.x)
}

fun pythagoreanTheorem(a: Double, b: Double): Double {
    return sqrt(a.pow(2) + b.pow(2))
}



